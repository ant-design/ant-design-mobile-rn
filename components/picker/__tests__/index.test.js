/**
 * @jest-environment jsdom
 */
import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'
import { Picker, Provider } from '../../'

jest.mock('@react-navigation/stack', () => ({
  generateKey: jest.fn(() => 123),
}))
describe('Picker', () => {
  describe('test cascade props', () => {
    it('when cascade={true} should be render MultiPicker', () => {
      const onChange = jest.fn()
      const onPress = jest.fn()

      // I cant resolve :(, same issue as https://github.com/enzymejs/enzyme/issues/1990 when:
      // const wrapper = mount(<TouchableHighlight><Text>1</Text></TouchableHighlight>)

      const data = [
        {
          label: '北京市',
          value: '11',
          children: [
            {
              label: '市辖区',
              value: '1101',
              children: [
                { label: '东城区', value: '110101' },
                { label: '西城区', value: '110102' },
              ],
            },
          ],
        },
      ]
      const wrapper = renderer.create(
        <Provider>
          <Picker data={data} cols={3} onChange={onChange}>
            <Text onPress={onPress} testID={'button'}>
              省市选择
            </Text>
          </Picker>
        </Provider>,
      )
      wrapper.root.findByType(Text).props.onPress()
      expect(wrapper).toMatchSnapshot()
    })

    // it('when cascade={false} should be render MultiPicker', () => {
    //   const seasons = [
    //     [
    //       {
    //         label: '2013',
    //         value: '2013',
    //       },
    //       {
    //         label: '2014',
    //         value: '2014',
    //       },
    //     ],
    //     [
    //       {
    //         label: '春',
    //         value: '春',
    //       },
    //       {
    //         label: '夏',
    //         value: '夏',
    //       },
    //     ],
    //   ];
    //   const { debug, container } = render(<PickerView
    //     data={seasons}
    //     cascade={false}
    //   />);
    //   debug();
    //   expect(container.getByDisplayValue('MultiPicker')).toBe(true);
    // });
  })

  // describe('test some events', () => {
  //   // fix: https://github.com/ant-design/ant-design-mobile-rn/issues/734
  //   it('fires onPickerChange event', () => {
  //     const dataFromIssue = [
  //       [
  //         { label: 'label1', value: 'value1' },
  //         { label: 'label2', value: 'value2' },
  //       ],
  //       [
  //         { label: 'label3', value: 'value3' },
  //         { label: 'label4', value: 'value4' },
  //       ],
  //     ];
  //     const onPickerChange = jest.fn();

  //     const wrapper = render(<Picker
  //       data={dataFromIssue}
  //       cols={2}
  //       cascade={false}
  //       onPickerChange={onPickerChange}
  //     >
  //       <Text testID={'button'}>省市选择</Text>
  //     </Picker>);

  //     wrapper.find(PopupCascader).prop('cascader').props.onValueChange(dataFromIssue[0][0]);
  //     expect(onPickerChange).toBeCalledWith(dataFromIssue[0][0]);
  //   });
  // });
})
