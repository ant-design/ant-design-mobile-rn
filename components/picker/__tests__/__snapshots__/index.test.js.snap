// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker test cascade props when cascade={true} should be render MultiPicker 1`] = `
Array [
  <View
    collapsable={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Text
        extra="请选择"
        onPress={[Function]}
        testID="button"
      >
        省市选择
      </Text>
    </View>
  </View>,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "zIndex": 1000,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "flex": 1,
          },
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "backgroundColor": "black",
                "opacity": 0.5,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#e7e7e7",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "flexGrow": 1,
                "height": 44,
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#0ae",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                  Object {},
                ]
              }
            >
              取消
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#666",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#0ae",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                  Object {},
                ]
              }
            >
              确定
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "北京市",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "11",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "市辖区",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "1101",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "东城区",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "110101",
                  },
                  Object {
                    "label": "西城区",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "110102",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
